python_modules/dagster/dagster/core/definitions/graph.py

```diff
@@ -478,6 +482,13 @@ def to_job(
         config_mapping = None
         partitioned_config = None
 
+        if partitions_def:
+            check.inst_param(partitions_def, "partitions_def", PartitionsDefinition)
+            check.invariant(
+                config is None, "Can't supply both the 'config' and 'partitions_def' arguments"
```

Comment:
It would be possible in the future to reconcile them, but they don't play well currently.  That's because the choice of partition key determines the run config.